policies.go

```diff
@@ -791,6 +791,22 @@ func (d *dcAwareRR) Pick(q ExecutableQuery) NextHost {
 	}
 }
 
+type ConvictionPolicy interface {
+	// Implementations should return `true` if the host should be convicted, `false` otherwise.
+	AddFailure(error error, host *HostInfo) bool
+	//Implementations should clear out any convictions or state regarding the host.
+	Reset()
+}
+
+//return true on any error
```

Comment:
Improve this comment, something like
`SimpleConvictionPolicy implements a ConvictionPolicy which convicts all hosts regardless of error`